### PR TITLE
[ConstraintElim] Add tests for llvm.abs

### PR DESCRIPTION
**Pull request by alexander-shaposhnikov**
This is a preparation for https://github.com/llvm/llvm-project/pull/73189

Test plan: ninja check-all

### CODE REVIEW

**Comment by fhahn** [comment]
Thanks for putting up the tests as separate PR! Could you also add tests with predicates other than `sge` (including some that cannot be simplified) and some unsigned ones as well?

**Comment by alexander-shaposhnikov** [comment]
done

File: llvm/test/Transforms/ConstraintElimination/abs.ll [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+;
+  %cmp_arg_nonnegative = icmp sge i32 %arg, 0
+  call void @llvm.assume(i1 %cmp_arg_nonnegative)
+  %abs = tail call i32 @llvm.abs.i32(i32 %arg, i1 true)
+  %abs_plus_one = add nuw i32 %abs, 1
+  %cmp = icmp uge i32 %abs_plus_one, %arg
+  ret i1 %cmp
+}
+
+define i1 @abs_plus_one_unsigned_greater_or_eual_cannot_be_simplified(i32 %arg) {
```
Comment (fhahn) [inline llvm/test/Transforms/ConstraintElimination/abs.ll:74]: typo: eual -> equal.

**Review by fhahn — COMMENTED** [review]
Thanks for the update! I think the last case missing is one where a compare gets simplified to false.

File: llvm/test/Transforms/ConstraintElimination/abs.ll (hunk 1 above)
Comment (alexander-shaposhnikov) [reply llvm/test/Transforms/ConstraintElimination/abs.ll]: Fixed

**Comment by alexander-shaposhnikov** [comment]
Address comments

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**